tools/bin/ariaDownload.py

```diff
@@ -307,6 +333,85 @@ def check_cookie_is_logged_in(cj):
+    for info in infos:
+        if info[0]: succeed.extend(info[0])
+        if info[1]: failed.extend(info[1])
+        if info[2]: skipped.extend(info[2])
+        tot_time += float(info[3])
+        tot_bytes += float(info[4])
+
+
+    log.info ('Successes: %d files, %s bytes', len(succeed), tot_bytes)
+    _ = [print ('\n\t-', sf['file'], f'{sf["size"]/1024**2:.2f}') for sf in succeed]
+
+    if skipped: log.info('Skipped: %d files', len(skipped))
+    _ = [print ('\n\t-', sf) for sf in skipped]
+
+    if failed: log.info('Failures: %d files', len(failed))
+    _ = [print ('\n\t-', sf) for sf in failed]
+
+    if succeed: log.info('Average Rate: %.2f MB/sec', (tot_bytes/1024.0**2)/tot_time)
+
+    if failed:
+        log.critical('We recommend rerunning the same ariaDownload command to address the %d failures', len(failed))
+    else:
+        log.info ('All files have been downloaded successfully')
+    return
+
+def status_plot(wd, rates=None, use_all=False):
+    """ Save plot after each chunk on each core; use_all shows all calls to script """
+    if len(rates) == 1: # in case all successful (for testing)
+        return
+    with open(op.join(wd, 'avg_rates.csv'), 'a') as fh:
+        fh.write(','.join([str(avg_rate) for avg_rate in rates]) + '\n')
+
+    timestamps, rates = [], []
+    with open(op.join(wd, 'avg_rates.csv'), 'r') as fh:
+        for line in fh:
+            dat = line.strip().split(',')[1:]
+            timestamps.extend([line.split(',')[0]]*len(dat)) # repeat for later idx
+            rates.extend (dat)
+    timestamps = np.array([float(t) for t in timestamps])
+    rates      = np.array([float(rate) for rate in rates])
+    if not use_all:
+        rates = rates[timestamps==timestamps.max()]
+    x         = range(len(rates))
+    fig, axes = plt.subplots()
+    axes.scatter(x, rates, color='k')
+    if len(x) > 0:
+        coeffs    = np.polyfit(x, rates, 1)
+        trend     = np.polyval(coeffs, x)
+        axes.plot(x, trend, 'r--', label=f'$\Delta$ {coeffs[0]:.3f} MB/sec/prod')
+
+    axes.set_xlabel('ARIA Product Index')
+    axes.set_ylabel('MB/sec')
+    axes.set_title(f'Last Update: {str(datetime.now())}')
+    axes.legend();
+    kws = dict(dpi=150, bbox_inches='tight', pad_inches=0.025, transparent=False)
+    fig.savefig(op.join(wd, 'AvgDlSpeed'), **kws)
+    return
+
+class MiniLog(object):
+
+    """ Helper to capture stdout for plotting """
+    def __init__(self):
+        # https://stackoverflow.com/questions/14906764/how-to-redirect-stdout-to-both-file-and-console-with-scripting
+        self.terminal  = os.sys.stdout
+        self.avg_rates = []
+        return
+
+    def write(self, message):
+        self.terminal.write(message)
+        if 'Average Rate' in message:
+            self.avg_rates.append(float(message.strip().split()[-1].strip('MB/sec')))
+        return
+
+    def flush(self):
+        """Needed for python 3 compatibility"""
```

Comment:
Codacy found an issue: [First line should end with a period, question mark, or exclamation point (not 'y') (D415)](https://app.codacy.com/gh/aria-tools/ARIA-tools/pullRequest?prid=6310391)